include/tallies/cylinder_position_filter.hpp

```diff
@@ -0,0 +1,89 @@
+#ifndef CYLINDER_POSITION_FILTER_H
+#define CYLINDER_POSITION_FILTER_H
+
+#include <tallies/position_filter.hpp>
+#include <utils/position.hpp>
+
+#include <yaml-cpp/yaml.h>
+
+#include <array>
+
+class CylinderPositionFilter : public PositionFilter {
+ public:
+  enum class Orientation { X, Y, Z };
+
+  CylinderPositionFilter(Position origin, double radius, double dx, double dy,
+                         double dz, std::size_t nx = 1, std::size_t ny = 1,
+                         std::size_t nz = 1, Orientation z_ = Orientation::Z);
+
+  StaticVector3 get_indices(const Tracker& tktr);
+
+  StaticVector3 get_shape() {
+    StaticVector3 filter_shape{Nx_, Ny_, Nz_};
+
+    map_indexes(filter_shape);
+    return filter_shape;
+  }
+
+  double radius() const { return radius_; }
+  double inv_radius() const { return inv_radius_; }
+  double get_scaled_radius(const Position& r) const;
+
+  double z_min(const StaticVector3& indices) const;
+  double z_max(const StaticVector3& indices) const;
+
+  Position get_centre(const StaticVector3& indices) const;
```

Comment:
```suggestion
  Position get_center(const StaticVector3& indices) const;
```
I try to stick to US English spellings as much as possible, though this is also coming from someone who leaves comments in French ¯\\_(ツ)_/¯